client/src/app/detail.js

```diff
@@ -295,20 +261,6 @@ const Detail = observer((props) => {
                           {walletStore.defaultAddress === detailStore.owner && (
                             <Flex>
                               <Box>
-                                {!detailStore.isApproved && (
-                                  <Box className="wallet-buttons">
-                                    <Button.Outline
-                                      className="waifu-card-buttons"
-                                      onClick={approveAccount}
-                                    >
-                                      <span className="waifu-button-learnmore">
-                                        {" "}
-                                        Approve
-                                      </span>{" "}
-                                    </Button.Outline>
-                                  </Box>
-                                )}
-                                {detailStore.isApproved && (
                                   <center>

```

Comment:
@Phineas  doesn't this remove the ability for the user to approve?

Reply:
Excuse my lack of Web3 knowledge haha, I know approval is needed for the first time a user connects their wallet provider but it's not needed after they've already made the first approval, right? We should probably store their approval status in the state

Reply:
I've added it back and call it before the modal is shown just to make sure we run into any inconsistencies in my latest commit